### DISCUSSION TITLE
nerdctl save error: failed to get reader: content digest

### DISCUSSION THREAD

**Discussion by charmingdog** [containerd/nerdctl#1178] [Q&A] [answered]
Does anyone know what is the reason for this error：
```FATA[0000] failed to get reader: content digest sha256:2408cc74d12b6cd092bb8b516ba7d5e290f485d3eb9672efc00f0583730179e8: not foun```

![image](https://user-images.githubusercontent.com/20454889/176613702-95d1fc13-188f-46de-b342-5f5bf749cb92.png)

**Comment by junnplus** [comment]
How to reproduce?

**Reply by charmingdog** [reply]
I don‘t know what happened。After reinstall nerdctl，buildkit，containerd，now can ``` nerdctl save``` .But I get new error :
![image](https://user-images.githubusercontent.com/20454889/176645301-ee63eac6-e499-4925-8cba-4e00d739493a.png)
my dockerfile
```
FROM nginx:1.22.0-alpine
COPY GeoIP/GeoIPCity.dat /usr/share/GeoIP/ 
COPY GeoIP/GeoIP.dat /usr/share/GeoIP/ 
COPY GeoIP/GeoIPv6.dat /usr/share/GeoIP/ 
COPY conf/nginx.conf /etc/nginx/nginx.conf
```
what is the meanning of the info "failed to get reader from content store: content digest sha256:3231f33bb6e5dddceb8da8344438c39d123cb9f5cdbc664f9a64e42ff199db90: not found"

**Reply by junnplus** [reply]
What's your version of nerdctl and containerd?

**Reply by charmingdog** [reply]
![image](https://user-images.githubusercontent.com/20454889/176862426-c467c266-3188-4c88-8a03-38d442787b6b.png)

**Comment by fahedouch** [comment]
it looks like not fetch some layers. seems to be related to this https://github.com/containerd/nerdctl/issues/951 . As @junnplus said plz share your nerdctl version and try use the latest version

**Reply by charmingdog** [reply]
After cleaning the environment and reinstalling nerdctl ,buildkitd and containerd ,both versions 1.21.0 and 1.20.0 of nerdctl work fine
The clean operation is as follows：
```
systemctl stop buildkit.service
systemctl stop containerd.service
apt remove containerd containerd.io
apt purge containerd containerd.io
rm -rf /var/lib/containerd
rm -rf /var/lib/buildkit
rm -rf /var/run/containerd
rm -rf /var/run/buildkit
```
Maybe some previous operations make buildkit environment unclean.
Before cleaning，when building nginx images change "FROM nginx:1.22.0-alpine"  to "FROM nginx:1.21.0-alpine" can work fine
when building alpine ,change "FROM alpine:3.16" to "FROM alpine:3.15" can work fine.Very Stranger！Like something wrong with alpine:3.16

**Comment by charmingdog** [comment, marked as answer]
```
systemctl stop buildkit.service
systemctl stop containerd.service
apt remove containerd containerd.io
apt purge containerd containerd.io
rm -rf /var/lib/containerd
rm -rf /var/lib/buildkit
rm -rf /var/run/containerd
rm -rf /var/run/buildkit

```
Clean the environment and reinstall nerdctl ,buildkitd and containerd ,then everything is ok.
